sky/shell/utils/countdown_latch.cc

```diff
@@ -0,0 +1,28 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/shell/utils/countdown_latch.h"
+
+namespace sky {
+namespace shell {
+
+CountdownLatch::CountdownLatch(size_t count) : count_(count) {}
+
+CountdownLatch::~CountdownLatch() = default;
+
+void CountdownLatch::Wait() {
+  if (count_ > 0) {
+    std::unique_lock<std::mutex> lock(cv_mutex_);
+    cv_.wait(lock, [&] { return count_ == 0; });
+  }
+}
+
+void CountdownLatch::CountDown() {
+  if (--count_ == 0) {
```

Comment:
Yes. http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith

I can also not use std::atomic and instead lock for the decrement using cv_mutex_ if that makes it clearer.